### DISCUSSION TITLE
i forcefully removed commits but i want those code now how can i retrive those

### DISCUSSION THREAD

**Discussion by TheDeCOderP** [community/community#138937] [Code Security]
### Select Topic Area

General

### Body

i want forcefully removed commits code it is possible to get those code ?

**Comment by phuyalgaurav** [comment]
I think you can use git reflog, heres how:
1. Type this command on the repo
`git reflog`

2. Find the commit hash from the list that corresponds to the commit you want to restore.
Once you find it, you can check it out or reset to that commit:
`git checkout <commit-hash>`
Or
`git reset --hard <commit-hash>`

Gir reflog only works after a small amount of time, so do it quick.

**Comment by ghostinhershell** [comment]
Hi @TheDeCOderP, thanks so much for writing in. You can learn more about restoring deleted repositories here: https://docs.github.com/en/repositories/creating-and-managing-repositories/restoring-a-deleted-repository#about-repository-restoration.
